--- ./Configurations/15-android.conf	2019-05-28 09:29:42.380247602 -0700
+++ ./Configurations/15-android.conf-patched	2019-05-28 09:30:41.395106783 -0700
@@ -97,7 +97,7 @@
                 (my $tritools   = $triarch) =~ s/(?:x|i6)86(_64)?-.*/x86$1/;
                 $cflags .= " -target $tridefault "
                         .  "-gcc-toolchain \$($ndk_var)/toolchains"
-                        .  "/$tritools-4.9/prebuilt/$host";
+                        .  "/llvm/prebuilt/$host";
                 $user{CC} = "clang" if ($user{CC} !~ m|clang|);
                 $user{CROSS_COMPILE} = undef;
                 if (which("llvm-ar") =~ m|^$ndk/.*/prebuilt/([^/]+)/|) {
